modules/core/src/lighting/light-source.js

```diff
@@ -1,37 +1,64 @@
 import {Vector3} from 'math.gl';
-import ScenegraphNode from '../scenegraph/scenegraph-node';
+import {uid} from '../utils';
 
 // default light source parameters
-const DEFAULT_LIGHT_POSITION = [0.0, 0.0, 1.0];
-const DEFAULT_LIGHT_DIRECTION = [0.0, 0.0, -1.0];
-const DEFAULT_LIGHT_INTENSITY = 1.0;
 const DEFAULT_LIGHT_COLOR = [255, 255, 255];
+const DEFAULT_LIGHT_INTENSITY = 1.0;
 const DEFAULT_ATTENUATION = [0, 0, 1];
 
-class LightSource extends ScenegraphNode {
-  constructor(props) {
-    super(props);
-    const {color = DEFAULT_LIGHT_COLOR, intensity = DEFAULT_LIGHT_INTENSITY} = props;
+const DEFAULT_LIGHT_DIRECTION = [0.0, 0.0, -1.0];
+
+const DEFAULT_LIGHT_POSITION = [0.0, 0.0, 1.0];
+
+// glTF lights reference:
+// https://github.com/KhronosGroup/glTF/tree/master/extensions/2.0/Khronos/KHR_lights_punctual
+
+class Light {
+  constructor(props = {}) {
+    this.id = props.id || uid('light');
+    const {color = DEFAULT_LIGHT_COLOR} = props;
     this.color = color;
+    const {intensity = DEFAULT_LIGHT_INTENSITY} = props;
     this.intensity = intensity;
   }
+
+  // PRIVATE
+
+  // Helper: Extracts attenuation from either `props.attenuation`` or `props.intensity``
+  // Supports both sophisticated light model and the classic intensity prop
+  _getAttenuation(props) {
```

Comment:
put attenuation function in base light class is kind of wired, because it won't apply to ambient and directional light sources

Reply:
Very true. What doesn't come across in this PR is that it also applies to `SpotLight`s ( I cut those from this PR and placed in a follow up glTF integrartion PR).

Alternatives could be to duplicate the method in Point and Spot, or create one more base class, or perhaps make SpotLight inherit from PointLight:

```
             Light
       /       |       \
Ambient  Directional AttenuatedLight
                      /             \
                 PointLight        SpotLight
```

```
             Light
       /       |       \
Ambient  Directional PointLight
                          \
                        SpotLight
```

Reply:
Moved into PointLight. SpotLight can inherit from PointLight so it will work nicely.

Reply:
Make SpotLight inherit from PointLight is a better solution, because SpotLight really is just a PointLight with limited emitting directions.